asreview/hdf5_logging.py

```diff
@@ -0,0 +1,259 @@
+import json
+from datetime import datetime
+from pathlib import Path
+
+import h5py
+import numpy as np
+import itertools
+from asreview.settings import ASReviewSettings
+
+
+def _append_to_dataset(name, values, g, dtype):
+    if name not in g:
+        g.create_dataset(name, (len(values),), dtype=dtype, maxshape=(None,),
+                         chunks=True)
+    else:
+        g[name].resize((len(g[name])+len(values),))
+    dataset = g[name]
+    dataset[len(g[name])-len(values):] = values
+
+
+def _dict_from_group(g):
+    g_dict = {}
+    for key, value in g.attrs.items():
+        g_dict[key] = value
+    for key, value in g.items():
+        if isinstance(value, h5py.Dataset):
+            if value.shape is not None:
+                new_arr = np.array(value)
+                if str(new_arr.dtype).startswith('|S'):
+                    new_arr = np.array(new_arr, dtype=str)
+                g_dict[key] = new_arr.tolist()
+        else:
+            g_dict[key] = _dict_from_group(value)
+    return g_dict
+
+
+def _result_group(f, query_i):
+    try:
+        g = f[f'/results/{query_i}']
+    except KeyError:
+        g = f.create_group(f'/results/{query_i}')
+        g.attrs['creation_time'] = np.string_(datetime.now())
+    return g
+
+
+class HDF5_Logger(object):
```

Comment:
Naming conventions

Reply:
I thought HFD5Logger was a bit ugly, but that's okay I suppose.